src/Solana/Address.cpp

```diff
@@ -53,3 +55,76 @@ Address addressFromValidatorSeed(const Address& fromAddress, const Address& vali
     Data hash = TW::Hash::sha256(extended);
     return Address(PublicKey(hash, TWPublicKeyTypeED25519));
 }
+
+/*
+ * Check if given address is on ed25519 curve,
+ * Based on solana code, create_program_address()
+ * if curve25519_dalek::edwards::CompressedEdwardsY::from_slice(hash.as_ref())
+ * https://github.com/solana-labs/solana/blob/master/sdk/program/src/pubkey.rs#L135
+ * https://github.com/solana-labs/solana/blob/master/sdk/program/src/pubkey.rs#L153
+ */
+bool Address::isValidOnCurve(const Data& data) {
```

Comment:
from https://github.com/dalek-cryptography/curve25519-dalek/blob/master/src/edwards.rs#L486, it compressed y point, we probably can update code here: https://github.com/trustwallet/wallet-core/blob/master/src/PublicKey.cpp#L81

`isValidOnCurve` seems not a good name

Reply:
I've added this check to PublicKey::isValid().  May be a bold move, PublicKey ctor will throw with an invalid value.